float64 D1
float64 D2
float64 D3
float64 D4